### DISCUSSION TITLE
Proposal for new edittemplate layout

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6416] [Ideas]
Many years ago, I [proposed](https://github.com/Jermolene/TiddlyWiki5/issues/506) a new edit view layout. It was perhaps too radical even if I still think it is valid.

Here is instead a digestible part of that proposal but in an updated form. It concerns relocating the **type** field, mainly because the current placement causes asymmetry in the whole layout. My *feng shui* master says the following is much better. He is very wrinkled and eats plankton so he is worth listening to:

<img width="766" alt="image" src="https://user-images.githubusercontent.com/4534056/150683052-9e6f1432-f086-4842-9267-259d6289608c.png">

As you can see, one aspect is to no longer make it a textfield/combobox to type into, but instead only a dropdown to select from. Much cleaner UI and it is reasonable UX because I dare say that most people *rarely* use the type field at all. The crowd in here is the absolute super users. Now *IF* a text area is still needed for exotic custom types that, for some reason, don't show in the listing then it could actually be inside the dropdown, at it's top.

One particularly positive side effect is that we solve the problem with the dropdown always going out of viewport, forcing you to scroll:

<img width="362" alt="image" src="https://user-images.githubusercontent.com/4534056/150682101-eb589662-1dcb-4dd8-92b8-7b2826d47254.png">

Actually, it could be *even* cleaner: Just like currently, there is no need to at all mention the default type. So it could merely show the dropdown button and just maybe the label. If a non-default is selected then it makes sense to show it. 

And inside the dropdown, there could be an indicator next to the default indicating that it, indeed, is the selected one. (Maybe we should use the *chooser* interface? I.e the UI used in the controlpanel to choose e.g palette or theme.)

**Comment by pmario** [comment]
I do like the proposal. 

>Now IF a text area is still needed for exotic custom types that, for some reason, don't show in the listing then it could actually be inside the dropdown, at it's top.

Instead of a text field, we may be able to add a "Add your own" ... Similar to the "snap tool" as seen below. ... BUT we will need to add some help text that tells the users they should only add valid [MIME types](https://www.iana.org/assignments/media-types/media-types.xhtml). 

![grafik](https://user-images.githubusercontent.com/374655/150696560-7e6f573a-2211-4568-ae94-8ecc1c74c407.png)

**Reply by twMat** [reply]
@pmario - good idea.

**Comment by twMat** [comment]
The interest in this is surprisingly low. Maybe I'm posting in the wrong forum.

Anyway, here is another proposal for UI simplification, of the [editor tools toolbar](https://github.com/Jermolene/TiddlyWiki5/issues/6515). 

Anyone with thoughts, feel free to chime in. (Or better yet, make a PR which is something I can't do for code.)

**Reply by Jermolene** [reply]
I welcome your work on the edit template, and I'm sorry that it hasn't borne much fruit in the core yet. It may well be best to float ideas and get feedback on talk.tiddlywiki.org.
